core/services/vrf/listener_v1.go

```diff
@@ -344,7 +349,16 @@ func (lsn *listenerV1) ProcessRequest(req request) bool {
 	// 4. The eth node sees the request reorg and tells us about it. We do our fulfillment
 	// check and the node says its already fulfilled (hasn't seen the fulfillment reorged yet),
 	// so we don't process the request.
-	callback, err := lsn.coordinator.Callbacks(nil, req.req.RequestID)
+	// Subtract 5 since the newest block likely isn't indexed yet and will cause "header not
+	// found" errors.
+	currBlock := new(big.Int).SetUint64(lsn.getLatestHead() - 5)
+	m := bigmath.Max(req.confirmedAtBlock, currBlock)
+	ctx, cancel := context.WithTimeout(context.Background(), callbacksTimeout)
```

Comment:
The idiomatic way would be to do a `defer cancel()` right after this line, instead of calling `cancel()` directly.

Reply:
yes true but I'm pretty sure there's some logic below that we wouldn't want to trigger a deadline exceeded error with e.g `lsn.pipelineRunner.Run`, if that stuff does not have much of a latency then I suppose it's ok, otherwise I'd rather call the cancel here than defer it after the subsequent logic.

Reply:
That's not going to happen because that code doesn't take in this context that you created, so it will be unaffected by any deadlines you set. There's no implicit behavior incurred by creating a context - you have to pass it around for your code to actually use it.

If the pipeline runner _did_ take this context that you created, calling `cancel()` and then passing it in will actually make things fail immediately with an error like "context canceled", which would not even give your code a chance to run. That's why it's usually good practice to scope contexts as closely as possible and not use the same context for multiple operations if you can help it.

Reply:
Makes sense, I was worried that deferring the cancel would cause an error eventually but I suppose the code using it would already be bumped off the heap at that point.